addons/purchase/models/account_invoice.py

```diff
@@ -79,6 +79,11 @@ def _onchange_purchase_auto_complete(self):
     @api.onchange('partner_id', 'company_id')
     def _onchange_partner_id(self):
         res = super(AccountMove, self)._onchange_partner_id()
+        delivery_partner_id = self._get_invoice_delivery_partner_id()
+        new_fiscal_position_id = self.env['account.fiscal.position'].with_context(force_company=self.company_id.id).get_fiscal_position(self.partner_id.id, delivery_id=delivery_partner_id)
+        self.fiscal_position_id = self.purchase_id.fiscal_position_id \
+            or self.env['purchase.order'].browse(self.env.context.get('default_purchase_id')).fiscal_position_id \
```

Comment:
Why is this not required for the `sale.order`s?

Reply:
Because sale works differently and actually create the invoice, and not only display a view